src/com/carolinarollergirls/scoreboard/json/ScoreBoardJSONListener.java

```diff
@@ -276,21 +276,19 @@ private void processPosition(String path, Position p, boolean remove) {
 	}
 	
 	private void processPenaltyCodes(Settings s) {
-		PenaltyCodesManager pm = new PenaltyCodesManager(s);
 		updates.add(new WSUpdate("ScoreBoard.PenaltyCode", null));
-		
-        try {
-            PenaltyCodesDefinition penalties = pm.loadFromJSON();
-            for(PenaltyCode p : penalties.getPenalties()) {
-                
-                updates.add(new WSUpdate("ScoreBoard.PenaltyCode."+p.getCode(), p.CuesForWS(p)));
-            }
-            
-            
-            updates.add(new WSUpdate("ScoreBoard.PenaltyCode.?","Unknown"));
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
+		String file = s.get(PenaltyCodesManager.PenaltiesFileSetting);
+		if(file != null && !file.isEmpty()) {
+			try {
+				PenaltyCodesDefinition penalties = pm.loadFromJSON(file);
+				for(PenaltyCode p : penalties.getPenalties()) {
+					updates.add(new WSUpdate("ScoreBoard.PenaltyCode."+p.getCode(), p.CuesForWS(p)));
+				}
+				updates.add(new WSUpdate("ScoreBoard.PenaltyCode.?","Unknown"));
+			} catch (Exception e) {
```

Comment:
What exception is this catching? loadFromJSON doesn't have any checked exceptions.